File: app/src/main/java/org/jay/launchstarterexample/tasks/InitAMapTask.java
package org.jay.launchstarterexample.tasks;

import com.amap.api.location.AMapLocation;
import com.amap.api.location.AMapLocationClient;
import com.amap.api.location.AMapLocationClientOption;
import com.amap.api.location.AMapLocationListener;
import org.jay.appstarter.Task;

public class InitAMapTask extends Task {

    private AMapLocationClient mLocationClient = null;
    private AMapLocationClientOption mLocationOption = null;
    private AMapLocationListener mLocationListener = new AMapLocationListener() {
        @Override
        public void onLocationChanged(AMapLocation aMapLocation) {
            // 一些处理
        }
    };

    @Override
    public void run() {
        mLocationClient = new AMapLocationClient(mContext);
        mLocationClient.setLocationListener(mLocationListener);
        mLocationOption = new AMapLocationClientOption();
        mLocationOption.setLocationMode(AMapLocationClientOption.AMapLocationMode.Battery_Saving);
        mLocationOption.setOnceLocation(true);
        mLocationClient.setLocationOption(mLocationOption);
        mLocationClient.startLocation();
    }
}

File: app/src/main/java/org/jay/appstarterexample/tasks/InitAMapTask.kt
package org.jay.appstarterexample.tasks

import com.amap.api.location.AMapLocationClient
import com.amap.api.location.AMapLocationClientOption
import com.amap.api.location.AMapLocationListener
import com.amap.api.location.AMapLocation
import org.jay.appstarter.Task

/**
 * 高德 SDK 初始化任务
 */
class InitAMapTask : Task() {

    private var mLocationClient: AMapLocationClient? = null
    private var mLocationOption: AMapLocationClientOption? = null
    private val mLocationListener = AMapLocationListener {
        // 一些处理
    }

    override fun run() {
        mLocationClient = AMapLocationClient(mContext)
        mLocationClient!!.setLocationListener(mLocationListener)
        mLocationOption = AMapLocationClientOption()
        mLocationOption!!.locationMode = AMapLocationClientOption.AMapLocationMode.Battery_Saving
        mLocationOption!!.isOnceLocation = true
        mLocationClient!!.setLocationOption(mLocationOption)
        mLocationClient!!.startLocation()
    }
}